src/main/io/gps.c

```diff
@@ -1676,7 +2215,7 @@ static bool UBLOX_parse_gps(void)
         }
         GPS_svInfoReceivedCount++;
         break;
-    case MSG_SAT:
+    case CLSMSG(CLASS_NAV, MSG_NAV_SAT):
```

Comment:
Adding a guard here for manually configured GPS modules would save a fair bit of unnecessary processing:
```
            if (ARMING_FLAG(ARMED)) {
                break;
            }
```

Reply:
I think I understand - you are thinking that if the user is manually configured the GPS, and configured Sat messages to be on, then you'd like to stop processing those messages while armed, to reduce the CPU overhead?
What would happen to the buffer if we blocked the processing of the received data?

Reply:
This method seems to only fill the `GPS_svinfo_chn`, so that structure would stay the same. Otherwise, skipping the processing of the message does not affect the rest of the code, the next message should be parsed the same way.

Reply:
What I was thinking about was if we had packets arriving of this type, but then we broke out of the case statement before handling the data, would that matter?

Reply:
> What I was thinking about was if we had packets arriving of this type, but then we broke out of the case statement before handling the data, would that matter?

By the time it hits the `case CLSMSG(CLASS_NAV, MSG_NAV_SAT):` in `UBLOX_parse_gps( )`, it has already pulled all the bytes for that message out of the buffer.

So if there is a conditional that decides to ignore the message there, that's fine.
But we've already paid the price to parse the bytes into the message.
The low level parsing code in `gpsNewFrameUBLOX( )` has no idea which message type it's parsing, only what part of the frame (preamble, payload, checksum, etc.) it's working on.

If the user has manually configured their module to sent sat info all the time, that's their choice. It's probably not a good choice, but maybe they're doing it for some unexpected reason. Do we try to protect them from making bad manual config choices? imho, if they don't know what they're doing, leave it at auto config.

Reply:
> If the user has manually configured their module to sent sat info all the time, that's their choice. It's probably not a good choice, but maybe they're doing it for some unexpected reason

I personally leave it on (and have and will use using manual config :) so when I'm stuffing around with GNSS settings I have a better idea what's going on. It's only a rate 10 for me (1/sec), so the extra traffic on the wire is negligible.
It does not cause any issues, but would certainly appreciate lowering the CPU load wherever it is feasible.

> But we've already paid the price to parse the bytes into the message.

That's true; however, it does not mean we should pay the price for parsing every single satellite out of the message (32, that is)!
The processing code of this ln2226 moves fair amount of data for no reason: during flight no one can have a look at it.

I cannot come up with a good reason to process the message mid-flight, but for people with manual config, it actually helps to reduce the CPU load mid-flight (mind you, [most people setting their GPS up will not fully understand what they are doing](https://oscarliang.com/gps-settings-u-center/#Optimal-Settings). Protecting them from themselves is a good idea)

Reply:
Vanishingly few people manually configure their GPS modules.  In the past, it was much more necessary than with this PR, because the Auto config failed to achieve a reasonable setup.  But even then, most people didn't manually configure; they did a once only reconfiguration to get their hardware to work with Auto config.

Now that the Auto config delivers just about the best possible configuration that you could want, and does so automatically, the need for manual configuration is much less than before.  The user gets no sat info, and no NMEA messages cluttering up the buffer, and can set the rate reliably.

If a person wants to constrain the GNSS set, permanently, they can do that, and still let auto config set the update rate, disable the unwanted messages, and enable those you do want.

Hence, going forward, manual configuration would be pointless except for some very particular personal interest.  It's tough at this point to know what that might be.  

A person using manual config would need to know how to overwrite and save settings to a GPS.  If the small CPU overhead of checking sats is something that matters, they can use passthrough to turn Sats off, and set them permanently off, and while flying the OSD is all most people use.  They only need to see sat info while connected to a laptop, and while connected to a laptop, it is very quick and easy to turn sats on if needed, using a variety of tools.  At the field, the vast majority of pilots only use their OSD.  If they want more information, before takeoff, they need a laptop, and if they have a laptop and are messing with manual config, they can then temporarily enable sats and use uCenter or whatever to examine their sat connections.  Once they are happy, they can disable Sat info for flight.

To facilitate this uncommon application we require another condition to be checked, one which 99.9% of people will not benefit from since they are getting no sat data anyway.

At this point I'm not inclined to support this.

Reply:
> To facilitate this uncommon application we require another condition to be checked, one which 99.9% of people will not benefit from since they are getting no sat data anyway.

For autoconfig people this `switch-case` won't even be executed.
For the other 0.01% of people (we are still talking about thousands physical devices here), you argume that we indeed need to waste CPU time, even though we know with 100% certainty that the results of these operations are *completely* useless and inaccessible during flight. Interesting.

Reply:
Yes, because otherwise there is an unnecessary test in the code, one which isn't needed by the auto users, but you want to apply to them.